Docs: Overhaul of installation documentation

joelhans opened the pull request:
<!--
Describe the change in summary section, including rationale and degin decisions.
Include "Fixes #nnn" if you are fixing an existing issue.

In "Component Name" section write which component is changed in this PR. This
will help us review your PR quicker.

If you have more information you want to add, write them in "Additional
Information" section. This is usually used to help others understand your
motivation behind this change. A step-by-step reproduction of the problem is
helpful if there is no related issue.
-->

##### Summary

This is a fairly extensive overhaul of the installation area of the documentation. I had a few goals here:

- Minimize the length/complexity of the main installation page (`packaging/installer/`).
- Create separate pages for different installation methods and unique OS options and add them to the left-hand navigation
- Help direct users to the proper method for their OS/distro
- Promote our Docker/K8s options more heavily
- Eliminate Markdown linter errors as much as possible/within reason

A few notes:

- I didn't do much overhaul, or even basic editing, of each method page (minus kickstart/kickstart-static64). They're mostly copy-pastes from the main installation page. I have _another_ epic to fix those individually.
- I had to rebase multiple times with lots of complex merges, so there may be weird artifacts from that. I hope not, and I caught some, but there may be more.

##### Component Name

docs
packaging

##### Additional Information

One additional benefit of separating installation methods into separate documents is that we can more easily create an installation "wizard" where users input their OS/version and we use some JS to insert the recommended installation method onto the page directly. We haven't started working on this yet (too much JS for my taste), but it's on the roadmap!

prologic commented:
You _might_ want to fix this:

```sh
$ ./tests/installer/checksums.sh
00:00:09.451915  Validating documentation for kickstart.sh
00:00:09.452133  Invalid checksum for kickstart.sh in packaging/installer/README.md.
00:00:09.452264  checksum in docs: 
00:00:09.452300  current checksum: 952da7868b2c6b8819a7c600047400f5
The command "./tests/installer/checksums.sh" exited with 2.

```

prologic commented on packaging/installer/methods/alpine.md:
````diff
@@ -0,0 +1,29 @@
+
+```sh
+# install required packages
+apk add alpine-sdk bash curl zlib-dev util-linux-dev libmnl-dev gcc make git autoconf automake pkgconfig python logrotate
````
```suggestion
apk add alpine-sdk bash curl libuv-dev zlib-dev util-linux-dev libmnl-dev gcc make git autoconf automake pkgconfig python logrotate
```

prologic replied:
See: #7845

Ferroin commented on packaging/installer/UPDATE.md:
```diff
@@ -41,17 +45,23 @@ bash <(curl -Ss https://my-netdata.io/kickstart-static64.sh) --no-updates
 
-If a newer version of Netdata is not available from the source you got it, we suggest to uninstall the version you have and follow the [installation](README.md) instructions for installing a fresh version of Netdata.
+If a newer version of Netdata is not available from the source you got it, we suggest to uninstall the version you have
+and follow the [installation](README.md) instructions for installing a fresh version of Netdata.
```
They should never need to go through this, they should be going through the regular package manager based workflow for their system to update the package.

Ferroin commented on packaging/installer/methods/freebsd.md:
````diff
@@ -0,0 +1,17 @@
+# install Netdata in /opt/netdata
+cd netdata
+./netdata-installer.sh --install /opt
+```
````
We should probably mention somehwere here that there is an (unofficial AFAIK) Netdata package in the FreeBSD ports tree. Many users are likely to prefer that if they know about it, as ti integrates better with the rest of the system than our current installer.

Also, unless I'm mistaken, `kickstart.sh` _should_ work on FreeBSD, and that's what we should be recommending most of the time.

Ferroin commented on packaging/installer/methods/kickstart-64.md:
````diff
@@ -0,0 +1,82 @@
+command above, run the following:
+
+```bash
+[ "23e0f38dfb9d517be16393c3ed1f88bd" = "$(curl -Ss https://my-netdata.io/kickstart-static64.sh | md5sum | cut -d ' ' -f 1)" ] && echo "OK, VALID" || echo "FAILED, INVALID"
````
Because of this being moved, we need updates to CI (we check the hashes here to make sure they match the most recent state of the repository). This is why Tavis CI is failing right now on this PR.

If desired, I can push a commit to the PR branch to take care of this.

joelhans replied:
I'd appreciate that in this case. Maybe I can learn how to do this myself in a smaller, more contained PR.

Ferroin replied:
I'd be happy to provide instructions if you would like to do it yourself, it's not particularly hard to do (should be no more than two lines changed in a single script, and it's only updating file names), I just figured you might want to not have to worry about it right now.

Ferroin replied:
Correction, after looking at the relevant code again, it's a bit more involved than I thought (there are structural changes needed, since it assumes both the things we're checking checksums for have the checksums listed in the same file).

I'll make sure to document in the commit what needs to be updated if these get moved again in the future.

Ferroin replied:
Commit added to update the CI stuff.

For future reference, any further changes that need to be made because of moving these lines just need to modify the lines in `tests/installer/checksums.sh` near the end that start with `check_checksums` to point to the correct files for the respective kickstart scripts.

Ferroin commented on packaging/installer/methods/kickstart.md:
````diff
@@ -0,0 +1,65 @@
+run the following:
+
+```bash
+[ "0ae8dd3c4c9b976c4342c9fc09d9afae" = "$(curl -Ss https://my-netdata.io/kickstart.sh | md5sum | cut -d ' ' -f 1)" ] && echo "OK, VALID" || echo "FAILED, INVALID"
````
Same as the above comment, this being moved needs a CI change, and I'll be happy to push a commit to handle that if desired.

Ferroin commented:
Fascinating, even though I was the one who pushed the most recent commit here, it still lists Joel as the one requesting Costa's review due to ownership of the file that was changed.

Also, LGTM appears to have lost it's mind (again)...

joelhans commented:
Hey @Ferroin — Thanks for your comments! Let me play around with the text here to see if I can promote the packages a bit more, while still making `kickstart.sh` the primary method. I think we need to promote that primarily still, as it's the recommended installation method in a few other places:

- The netdata.cloud homepage
- Our main README.md
- The step-by-step tutorial
- ... and a few other places, too

I'll ping you when I finish up what I have in mind, but we may need to have a bigger conversation around the process of eventually making packages _the_ recommended method.

Also, props to @prologic for the git tips that should make the _next_ massive PR I do with lots of moved/new files a little easier for everyone to follow.

Ferroin commented:
> Hey @Ferroin — Thanks for your comments! Let me play around with the text here to see if I can promote the packages a bit more, while still making `kickstart.sh` the primary method. I think we need to promote that primarily still, as it's the recommended installation method in a few other places.

Agreed. Long term the plan is to have `kickstart.sh` preferentially set up the system to use our repos and then install the packages instead of building locally, so even then it should be our primary recommendation, but making it easy for people to figure out how to use the packages by hand is important, as some people (rightfully) won't want to download and run a random shell script off the internet.

joelhans commented:
> > Hey @Ferroin — Thanks for your comments! Let me play around with the text here to see if I can promote the packages a bit more, while still making `kickstart.sh` the primary method. I think we need to promote that primarily still, as it's the recommended installation method in a few other places.
> 
> Agreed. Long term the plan is to have `kickstart.sh` preferentially set up the system to use our repos and then install the packages instead of building locally, so even then it should be our primary recommendation, but making it easy for people to figure out how to use the packages by hand is important, as some people (rightfully) won't want to download and run a random shell script off the internet.

Okay. That's really good for me to know for sure—I remember you mentioning something like that in Athens, but I wasn't sure if that was the actual long-term plan.

joelhans commented:
@Ferroin @prologic I've addressed both of your issues and rebased to master while implementing the latest fix (missing hyphen in `--disable-telemetry`) in both the `kickstart.md` and `kickstart-static64.md` files.

@Ferroin — Regarding the rest of your concerns around the updating/uninstalling documents, I think I should create separate issues/PRs for these and we can address them individually. This will be a good opportunity for me to finally dive into those documents, test some methods out, and get them up-to-date.

We should probably coordinate the approval process next week. I don't want to block your work in improving the installation docs/scripts, but I can see the rebasing and dismissing approvals causing issues if we're all touching `packaging/installer/README.md` a lot. Let's all chat on Monday to see what we can figure out.

prologic commented:
Btw.. I spoke with @joelhans about this PR and I'm a bit worried we're reviewing things he moved around and restructured -- i.e: reviewing things already previously merged/accepted before. Not that that's a bad thing -- but it complicated this PR's review :)
